resources/benchmark-runner.mjs

```diff
@@ -227,50 +227,50 @@ export class BenchmarkRunner {
             if (this._client && this._client.willRunTest)
                 this._client.willRunTest(suite, test);
 
-            setTimeout(() => {
-                this._runTest(suite, test, this._page, (syncTime, asyncTime) => {
-                    const suiteResults = this._measuredValues.tests[suite.name] || {tests:{}, total: 0};
-                    const total = syncTime + asyncTime;
-                    this._measuredValues.tests[suite.name] = suiteResults;
-                    suiteResults.tests[test.name] = {tests: {'Sync': syncTime, 'Async': asyncTime}, total: total};
-                    suiteResults.total += total;
-
-                    if (this._client && this._client.didRunTest)
-                        this._client.didRunTest(suite, test);
-
-                    resolve();
-                });
-            }, 0);
+             window.requestAnimationFrame(
+                this._runTest.bind(this, suite, test, this._page, resolve));
         });
     }
 
     // This function ought be as simple as possible. Don't even use Promise.
-    _runTest(suite, test, page, callback)
+    _runTest(suite, test, page, test_done_callback)
     {
-        const now = window.performance && window.performance.now ? () => window.performance.now() : Date.now;
-
         this._writeMark(suite.name + '.' + test.name + '-start');
-        let startTime = now();
+        const startTime = window.performance.now();
         test.run(page);
-        let endTime = now();
+        const endTime = window.performance.now();
         this._writeMark(suite.name + '.' + test.name + '-sync-end');
 
         const syncTime = endTime - startTime;
+        this._measureTestAsyncTime(suite, test, syncTime, test_done_callback);
+    }
 
-        startTime = now();
+    _measureTestAsyncTime(suite, test, syncTime, test_done_callback)
+    {
+        const startTime = window.performance.now();
         setTimeout(() => {
```

Comment:
Isn't it possible that test.run() triggered some promises, which have their callbacks run at the microtask checkpoint, which happens _after_ this setTimeout has been called. And those callbacks might trigger some setTimeout(0) stuff. All that, maybe non-insignificant work, would happen after the timeout above runs.

We could have two rAF callbacks. First one runs sync tests, and since microtasks run right after it, all the promises are handled there too. Then the second rAF callback would be capture the end time of the sync testing and call setTimeout(, 0). That would push the setTimeout of the testing framework to happen after the ones triggered by run.test()